it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/storage/GcsResourceManagerTest.java

```diff
@@ -158,6 +170,36 @@ public void testUploadArtifactInvalidLocalPath() {
         IOException.class, () -> gcsClient.uploadArtifact(ARTIFACT_NAME, "/" + UUID.randomUUID()));
   }
 
+  @Test
+  public void testGenerateBucketName_validInput() {
+    String testClassName = "MyTestClass";
+    String runId = "1234567890";
+    String expectedBucketName = "mytestclass-1234567890";
+    String actualBucketName = GcsResourceManager.generateBucketName(testClassName, runId);
+    assertEquals(expectedBucketName, actualBucketName);
+  }
+
+  @Test
+  public void testGenerateBucketName_withSpecialCharactersAndLongName() {
+    String testClassName = "My.Test.Class.With.A.Very.Long.Name";
+    String runId = "1234567890123456789012345678901234567890";
+    String expectedBucketName =
```

Comment:
nit: The comments should come on the line before. It would be easier to read the full comment that way.

Reply:
Dne